Library/Homebrew/test/version_spec.rb

```diff
@@ -234,14 +234,14 @@
 
   describe "::parse" do
     it "returns a NULL version when the URL cannot be parsed" do
-      expect(Version.parse("http://example.com/blah.tar")).to be_null
-      expect(Version.parse("foo")).to be_null
+      expect(described_class.parse("http://example.com/blah.tar")).to be_null
+      expect(described_class.parse("foo")).to be_null
     end
   end
 
   describe "::detect" do
     matcher :be_detected_from do |url, specs = {}|
-      detected = Version.detect(url, specs)
+      detected = described_class.detect(url, specs)
```

Comment:
This here fails because it is in a different scope, I think.

Reply:
@reitermarkus Any idea if there's a way to somehow have it work?

Reply:
This works:

```ruby
      match do |expected|
        @detected = described_class.detect(url, specs)
        @detected == expected
      end

      failure_message do |expected|
        message = <<~EOS
          expected: %s
          detected: %s
        EOS
        format(message, expected, @detected)
      end
```